content/gsoc/2022.md

```diff
@@ -0,0 +1,600 @@
+Participant should know C as well as have the experience of debugging with GDB or LLDB.
+
+### Difficulty
+Hard
+
+### Benefits for the participant
+Participant will understand how high-level features of debuggers work as well as gain skills in the field of software architecture of a large, modular C project.
+
+### Assess requirements for midterm/final evaluation
+- 1st term: `debuginfod` and source line information refactoring are implemented
+- Final term: Integration of variable information with the analysis and debug printing is implemented
+
+### Mentors
+ - xvilka
+ - thestr4ng3r
+ - ret2libc
+
+### Links/Resources
+- [Loading debug information from `debuginfod`](https://github.com/rizinorg/rizin/issues/1285)
+- [Unify code of source information access for DWARF, PDB, dSYM](https://github.com/rizinorg/rizin/issues/907)
+- [Debuginfod](https://sourceware.org/elfutils/Debuginfod)
+- [Debian Debuginfod](https://wiki.debian.org/Debuginfod)
+- [Fedora Debuginfod](https://fedoraproject.org/wiki/Debuginfod)
+- [DWARF-labeled issues](https://github.com/rizinorg/rizin/labels/DWARF)
+- [PDB-labaled issues](https://github.com/rizinorg/rizin/labels/PDB)
+
+## Thread-safety and multithreading (Medium project)
+
+Currently Rizin is not thread safe completely internally and as a library for multithreaded
+application. The goal of this project is to eliminate global states and use contexts,
+eliminate singletons, e.g. RzCons, and use thread-safe external functions and dependencies.
+
+### Task
+1. Migrate from thread-unsafe system and external dependencies
+2. Eliminate global state inside RzCons and use of the singleton
+3. Make RzBin thread-safe
+4. Make RzAnalysis thread-safe
+5. Make RzCore thread-safe
+6. Add tests for using multiple RzCore and RzAnalysis instances
+7. Parallelize some of the RzAnalysis function using the threading API
+
+### Skills
+Participant should know C as well as have the experience of developing multithreaded applications.
+
+### Difficulty
+Hard
+
+### Benefits for the participant
+Participant will understand the hurdles of multithreaded programming, data syncronization, locks and debugging of such code.
+
+### Assess requirements for midterm/final evaluation
+- 1st term: `RzTypes` module exists and contains all relevant code.
+- Final term: Tree-sitter based C parser is implemented and integrated into the analysis framework.
+
+### Mentors
+ - xvilka
+ - thestr4ng3r
+ - wargio
+
+### Links/Resources
+- [Migrate from `wcstombs()` function since it's not thread-safe](https://github.com/rizinorg/rizin/issues/1058)
+
+## Rewriting GPL-only code (Medium project)
+
+Currently some of the Rizin main features rely on the GPL-only code copied from binutils or GCC.
+The goal is to rewrite all this code from GPL-only to LGPL or any other less restrictive license.
+It is quite important for better adoption of Rizin as a library in other FOSS and commercial
+projects.
+
+## Tasks
+1. Rewrite C++ demangler to no use GPL-only code
```

Comment:
```suggestion
1. Rewrite C++ demangler to and remove the GPL code
```

Reply:
maybe implement the GNUv2 C++ demangler if too easy?

Reply:
Yes